src/emulator/functionsEmulator.ts

```diff
@@ -459,33 +457,35 @@ export class FunctionsEmulator implements EmulatorInstance {
         `No node binary for ${emulatableBackend.functionsDir}. This should never happen.`
       );
     }
-    const worker = this.invokeRuntime(
-      this.getBaseBundle(emulatableBackend),
-      {
-        nodeBinary: emulatableBackend.nodeBinary,
-        extensionTriggers: emulatableBackend.predefinedTriggers,
-      },
-      // Don't include user envs when parsing triggers.
-      {
-        ...this.getSystemEnvs(),
-        ...this.getEmulatorEnvs(),
-        FIREBASE_CONFIG: this.getFirebaseConfig(),
-        ...emulatableBackend.env,
-      }
-    );
-
-    const triggerParseEvent = await EmulatorLog.waitForLog(
-      worker.runtime.events,
-      "SYSTEM",
-      "triggers-parsed"
-    );
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    const parsedDefinitions = triggerParseEvent.data
-      .triggerDefinitions as ParsedTriggerDefinition[];
-
-    const triggerDefinitions: EmulatedTriggerDefinition[] =
-      emulatedFunctionsByRegion(parsedDefinitions);
 
+    let triggerDefinitions: EmulatedTriggerDefinition[];
+    if (emulatableBackend.predefinedTriggers) {
```

Comment:
Probably another TODO that this should be unified after extensions no longer use extensions.yaml for their resource definitions (e.g. we'll start using the normal path for newly written extensions and then in CLIv11 we'll break the old extension definitions)

Reply:
Yep. I think that' something I'll leave to Extensions team to think through 🤔